presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -419,6 +419,18 @@ private void invokeNoMoreSplitsIfNecessary()
                     false,
                     true));
 
+            if (AcidUtils.isFullAcidTable(table.getParameters())) {
+                /**
+                 * From Hive version >= 3.0, delta/base files will always have file '_orc_acid_version' with value >= '2'.
+                 */
+                Path baseOrDeltaPath = directory.getBaseDirectory() != null ? directory.getBaseDirectory() :
+                        directory.getCurrentDirectories().size() > 0 ? directory.getCurrentDirectories().get(0).getPath() : null;
+
+                if (baseOrDeltaPath != null && AcidUtils.OrcAcidVersion.getAcidVersionFromMetaFile(baseOrDeltaPath, fs) < 2) {
+                    throw new PrestoException(NOT_SUPPORTED, String.format("Hive 2.0 versioned transactional tables are not supported, Please run major compaction in Hive 3.0"));
```

Comment:
- `format` is unnecessary. 
- we don't know whether this is Hive 2 or Hive 1. Maybe 
  ```
  "Hive transactional tables are supported with Hive 3.0 and only after a major compaction has been run"
  ```

Reply:
Rephrased a little, please take a look.

Reply:
My proposal was:

> Hive transactional tables are supported with Hive 3.0 and only after a major compaction has been run

Currently it's:

> Hive version <= 2.0 transactional tables are not supported, Please run major compaction in Hive 3.0



The problem with current wording is that exception message should be "declarative", stating the problem.

@electrum can you suggest wording here?